core/store/migrate/migrations/0196_add_txhash_index_evm_logs.sql

```diff
@@ -0,0 +1,5 @@
+-- +goose Up
+create index evm_logs_idx_tx_hash on evm.logs (tx_hash);
```

Comment:
I verified it on my local Postgres instance, around 4k rows matching (rather pessimistic scenario)

Before
```sql
 Sort  (cost=30843.63..30843.63 rows=1 width=277) (actual time=52.694..52.870 rows=3660 loops=1)
   Sort Key: (ROW(block_number, log_index))
   Sort Method: quicksort  Memory: 1069kB
   ->  Bitmap Heap Scan on logs  (cost=21187.40..30843.62 rows=1 width=277) (actual time=47.864..49.920 rows=3660 loops=1)
         Recheck Cond: ((evm_chain_id = '1'::numeric) AND (event_sig = '\xaaaa'::bytea))
         Filter: (tx_hash = '\xbbcc'::bytea)
         Rows Removed by Filter: 10000
         Heap Blocks: exact=3503
         ->  Bitmap Index Scan on idx_evm_logs_by_timestamp  (cost=0.00..21187.40 rows=3348 width=0) (actual time=43.163..43.163 rows=13660 loops=1)
               Index Cond: ((evm_chain_id = '1'::numeric) AND (event_sig = '\xaaaa'::bytea))
 Planning Time: 0.380 ms
 Execution Time: 53.353 ms
(12 rows)
```

After adding index
```sql
 Sort  (cost=49.62..49.62 rows=1 width=277) (actual time=11.469..11.642 rows=3660 loops=1)
   Sort Key: (ROW(block_number, log_index))
   Sort Method: quicksort  Memory: 1069kB
   ->  Index Scan using evm_logs_idx_tx_hash on logs  (cost=0.42..49.61 rows=1 width=277) (actual time=0.067..7.170 rows=3660 loops=1)
         Index Cond: (tx_hash = '\xbbcc'::bytea)
         Filter: ((evm_chain_id = '1'::numeric) AND (event_sig = '\xaaaa'::bytea))
         Rows Removed by Filter: 7340
 Planning Time: 0.899 ms
 Execution Time: 12.132 msws)
```

Test with index matching all columns used here
```sql
create index evm_logs_idx_tx_hash_long on evm.logs (evm_chain_id, event_sig, tx_hash);
```


```sql
 Sort  (cost=8.58..8.59 rows=1 width=277) (actual time=9.194..9.365 rows=3660 loops=1)
   Sort Key: (ROW(block_number, log_index))
   Sort Method: quicksort  Memory: 1069kB
   ->  Index Scan using evm_logs_idx_tx_hash_long on logs  (cost=0.55..8.57 rows=1 width=277) (actual time=0.071..4.628 rows=3660 loops=1)
         Index Cond: ((evm_chain_id = '1'::numeric) AND (event_sig = '\xaaaa'::bytea) AND (tx_hash = '\xbbcc'::bytea))
 Planning Time: 0.417 ms
 Execution Time: 9.872 ms
(7 rows)
```

Reply:
```sql
   ->  Index Scan using evm_logs_idx_tx_hash on logs  (cost=0.42..49.61 rows=1 width=277) (actual time=0.067..7.170 rows=3660 loops=1)
```
versus 
```sql
   ->  Index Scan using evm_logs_idx_tx_hash_long on logs  (cost=0.55..8.57 rows=1 width=277) (actual time=0.071..4.628 rows=3660 loops=1)
```

Which gives us
`cost=0.42..49.61` versus `cost=0.55..8.57` when all fields match. Scanning when all columns are matching is faster. If the index is applied only on a single column, Postgres planner picks up the best index but has to use additional filtering.  Of course, real latency depends on the data distribution here, if `tx_hash` is rather unique and after filtering by `tx_hash`, the result set will be extremely small then probably your index will be enough, otherwise, I think we should have 3 columns within the index.

Reply:
On the other hand, it's hard for me to estimate what will be the cost of having 3 columns versus single column index

Reply:
The resulting logs from a single tx hash should be <100 in almost all cases, with most probably <5 entries found, as we only subscribe to a small set of topics.